srtp/srtp.c

```diff
@@ -79,32 +79,32 @@ srtp_debug_module_t mod_srtp = {
 #define uint32s_in_rtcp_header 2
 #define octets_in_rtp_extn_hdr 4
 
-static srtp_err_status_t srtp_validate_rtp_header(void *rtp_hdr,
-                                                  int *pkt_octet_len)
+static srtp_err_status_t srtp_validate_rtp_header(const void *rtp_hdr,
+                                                  int pkt_octet_len)
 {
-    srtp_hdr_t *hdr = (srtp_hdr_t *)rtp_hdr;
+    const srtp_hdr_t *hdr = (const srtp_hdr_t *)rtp_hdr;
     int rtp_header_len;
 
-    if (*pkt_octet_len < octets_in_rtp_header)
+    if (pkt_octet_len < octets_in_rtp_header)
         return srtp_err_status_bad_param;
 
     /* Check RTP header length */
     rtp_header_len = octets_in_rtp_header + 4 * hdr->cc;
     if (hdr->x == 1)
         rtp_header_len += octets_in_rtp_extn_hdr;
 
-    if (*pkt_octet_len < rtp_header_len)
+    if (pkt_octet_len < rtp_header_len)
         return srtp_err_status_bad_param;
 
     /* Verifing profile length. */
     if (hdr->x == 1) {
-        srtp_hdr_xtnd_t *xtn_hdr =
-            (srtp_hdr_xtnd_t *)((uint32_t *)hdr + uint32s_in_rtp_header +
-                                hdr->cc);
+        const srtp_hdr_xtnd_t *xtn_hdr =
+            (const srtp_hdr_xtnd_t *)((const uint32_t *)hdr +
+                                      uint32s_in_rtp_header + hdr->cc);
         int profile_len = ntohs(xtn_hdr->length);
         rtp_header_len += profile_len * 4;
         /* profile length counts the number of 32-bit words */
-        if (*pkt_octet_len < rtp_header_len)
+        if (pkt_octet_len < rtp_header_len)
```

Comment:
Wow. I'm surprised the compiler did not complain about this one.

Reply:
this was due to api change from int * to int